api/base/filters.py

```diff
@@ -20,53 +30,184 @@ def filter_queryset(self, request, queryset, view):
+
+    def _validate_operator(self, field, field_name, op):
+        """
+        Check that the operator and field combination is valid
+
+        :raises InvalidFilterComparisonType: If the query contains comparisons against non-date or non-numeric fields
+        :raises InvalidFilterMatchType: If the query contains comparisons against non-string or non-list fields
+        :raises InvalidFilterOperator: If the filter operator is not a member of self.COMPARISON_OPERATORS
+        """
+        if op not in set(self.MATCH_OPERATORS + self.COMPARISON_OPERATORS + (self.DEFAULT_OPERATOR, )):
+            valid_operators = self._get_valid_operators(field)
+            raise InvalidFilterOperator(value=op, valid_operators=valid_operators)
+        if op in self.COMPARISON_OPERATORS:
+            if not self._isinstance_any(field, self.COMPARABLE_FIELDS):
+                raise InvalidFilterComparisonType(
+                    parameter="filter",
+                    detail="Field '{0}' does not suport comparison operators in a filter.".format(field_name)
+                )
+        if op in self.MATCH_OPERATORS:
+            if not self._isinstance_any(field, self.MATCHABLE_FIELDS):
+                raise InvalidFilterMatchType(
+                    parameter="filter",
+                    detail="Field '{0}' does not support match operators in a filter.".format(field_name)
+                )
+
+    def _parse_date_param(self, field, field_name, op, value):
+        """
+        Allow for ambiguous date filters. This supports operations like findings Nodes created on a given day
+        even though Node.date_created is a specific datetime.
+
+        :return list<dict>: list of one (specific datetime) or more (date range) parsed query params
+        """
+        time_match = self.DATETIME_PATTERN.match(value)
+        if op != 'eq' or time_match:
+            return [{
+                'op': op,
+                'value': self.convert_value(value, field)
+            }]
+        else:  # TODO: let times be as generic as possible (i.e. whole month, whole year)
+            start = self.convert_value(value, field)
+            stop = start + datetime.timedelta(days=1)
+            return [{
+                'op': 'gte',
+                'value': start
+            }, {
+                'op': 'lt',
+                'value': stop
+            }]
+
+    def parse_query_params(self, query_params):
+        """Maps query params to a dict useable for filtering
+        :param dict query_params:
+        :return dict: of the format {
+            <resolved_field_name>: {
+                'op': <comparison_operator>,
+                'value': <resolved_value>
+            }
+        }
+        """
+        query = {}
+        for key, value in query_params.iteritems():
+            match = self.QUERY_PATTERN.match(key)
+            if match:
+                field_name = match.groupdict()['field'].strip()
```

Comment:
Define match.groupdict() once as a separate variable, since it is used multiple times.